utils/src/main/java/software/amazon/awssdk/utils/AttributeMap.java

```diff
@@ -83,10 +83,14 @@ public AttributeMap copy() {
     }
 
     @Override
-    public void close() {
-        attributes.values().forEach(v -> IoUtils.closeIfCloseable(v, null));
-        attributes.values().forEach(this::shutdownIfExecutorService);
-    }
+    public void close() { 
+     attributes.values().stream()
+             .filter(value -> !(value instanceof ExecutorService))
+             .forEach(v -> IoUtils.closeIfCloseable(v, null)); 
+     attributes.values().stream()
+             .filter(value -> value instanceof ExecutorService)
```

Comment:
Well, do a filter and later check again in shutdownIfExecutorService method is not a good deal. I would remove this filter or change the method.